cnxarchive/utils/ident_hash.py

```diff
@@ -29,6 +29,8 @@ class IdentHashError(Exception):
 
 class IdentHashSyntaxError(IdentHashError):
     """Raised when the ident-hash syntax is incorrect."""
+    def __init__(self, ident_hash):
+        self.ident_hash = ident_hash
```

Comment:
I've used `__str__` instead of `self.args`... :stuck_out_tongue:   I'll change it if it's not a good idea.  Exceptions will now look like this:

```
>>> raise IdentHashMissingVersion('8ac77e82-95d8-4a90-8b06-80a83bb8e636')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cnxarchive.utils.ident_hash.IdentHashMissingVersion: id=8ac77e82-95d8-4a90-8b06-80a83bb8e636
>>> raise IdentHashShortId('12345678', 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cnxarchive.utils.ident_hash.IdentHashShortId: id=12345678 version=1
>>> raise IdentHashSyntaxError('a@')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
cnxarchive.utils.ident_hash.IdentHashSyntaxError: ident_hash=a@
```